src/kinetics.F90

```diff
@@ -144,7 +144,7 @@ function force( this, vmr )
     real(musica_dk)              ::  force(size(vmr))    ! rate of change of each molecule
 
     !force = p_force( vmr, this%rates, this%number_density, this%rateConst )
-    call p_force( this%rateConst, vmr, this%environment%number_density_air, force)
+    call p_force( this%rateConst, vmr, this%environment(1)%number_density_air, force)
```

Comment:
I think `number_density_air` will have to be an array. I'm pretty sure the other approach would require `p_force()` to be an elemental function; it wouldn't get converted into an array.

I see now that I was using the wrong preprocessed files in the build. I will go through and fix the function calls missing the `ncell` argument.

Reply:
After switching to the correct preprocessed code, I think the remaining changes (inclusion of `ncells`) are best addressed by including your updates to the solver functions. Do you want to merge them in here? Or merge this PR to your branch and continue from there?

Reply:
Thanks @mattldawson . After reading your other replies, I think it may be better not to pass `ncell` as an argument, but infer it from the other argument like `Y`. This may avoid the revision required for `p_force` or other functions.

Reply:
And I would like to merge your PR first and then continue from there for the changes of solver functions.